src/main/java/com/faforever/client/chat/ChatController.java

```diff
@@ -215,6 +216,13 @@ private void joinChannel(String channelName) {
     chatService.joinChannel(channelName);
   }
 
+  @Subscribe
+  public void onFocusChannelTab(FocusChannelTabEvent focusChannelTabEvent) {
+    Platform.runLater(() -> {
+      tabPane.getSelectionModel().select(getOrCreateChannelTab(focusChannelTabEvent.getChannel()).getRoot());
+    });
+  }
+
```

Comment:
We are slowly converting to spring event Bus, so better use applicationService.post for new Events. Also this should be a NavigationEvent so it can be used also from the outside. Best would be an event that creates channel and focuses it, can be used in other contexts as well then

Reply:
It already auto creates a TabController which auto joins a channel, what do you mean?

Reply:
There is no ApplicationService in spring. Do you mean the one from JavaFX?

Reply:
Also just changing one post when there are multiple others in the same class doesn't make a lot of sense. Make an issue for it ;).

Reply:
He means `ApplicationEventPublisher`. I do agree we'd rather have all fixed at once instead of mixing things up. However, one issue to fix them all would result in a huge PR with a big potential for many conflicts. Therefore, I'd rather see us fix one location at a time when we're touching it anyway - especially when it's new code.

Reply:
This is not the right place to fix it, as this class publishes events to about 10 other classes. Also the Event I'm using extends NaviagteEvent, so if we wanna extend ApplicationEvent in that class, we're gonna have to change a lot.